daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -327,11 +328,17 @@ final class PState private (
     this.infoset = newParent
   }
 
-  def setVariable(vrd: VariableRuntimeData, newValue: DataValuePrimitive, referringContext: VariableRuntimeData): Unit = {
+  def setVariable(vrd: VariableRuntimeData, newValue: DataValuePrimitive, referringContext: ThrowsSDE): Unit = {
     variableMap.setVariable(vrd, newValue, referringContext, this)
     changedVariablesStack.top += vrd.globalQName
   }
 
+  def getVariable(vrd: VariableRuntimeData, referringContext: ThrowsSDE): DataValuePrimitive = {
+    val res = variableMap.readVariable(vrd, referringContext, this)
+    changedVariablesStack.top += vrd.globalQName
```

Comment:
This has no coverage, should our variable code be using this getVariable so that the stack is adjusted correctly, similar to setVariable? Seems inconsistent, and maybe a bug, if setVariable and getVariable are access in different ways?

Reply:
This was needed in the first version of IPv4 which had two transforms where the first one assigned a variable that had to be read (gotten) by the second. 

When I simplified IPv4 to a single transform, this was no longer needed. And check-digits doesn't need it either.

Neither one reads any variable. Just writes them. 

But there's nothing stopping a layer transform from wanting to read a variable, e.g, to get a parameter to the algorithm, e.g, a mode flag like compression algorithm to use, or variant of line-folding to use, etc. 

So I think we just need an example that exercises this call.